cmd/sealer/cmd/cluster/run.go

```diff
@@ -238,6 +245,89 @@ func runWithClusterfile(clusterFile string, runFlags *types.RunFlags) error {
 	return runClusterImage(imageEngine, cf, imageSpec, runFlags.Mode, runFlags.IgnoreCache)
 }
 
+func runLocalWithDocker(imageName string) error {
+	fmt.Println("imageName:", imageName)
+	arch := runtime.GOARCH
+	var dockerfilePath string
+
+	switch arch {
+	case "amd64":
+		dockerfilePath = "pkg/infra/container/imagecontext/base/Dockerfile"
+	case "arm64":
+		dockerfilePath = "pkg/infra/container/imagecontext/arm/Dockerfile"
+	default:
+		return fmt.Errorf("unsupported architecture: %s", arch)
+	}
+
+	err := buildDockerImage(imageName, dockerfilePath, arch)
+	if err != nil {
+		return err
+	}
+
+	err = runDockerContainer(imageName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildDockerImage(imageName, dockerfilePath, arch string) error {
+	dockerfile, err := ioutil.ReadFile(dockerfilePath)
+	if err != nil {
+		return err
+	}
+
+	newDockerfile := specifyImageName(imageName, dockerfile)
+
+	var (
+		tmpFile 		*os.File
+		errFile 		error
+	)
+
+	switch arch {
+	case "amd64":
+		tmpFile, errFile = ioutil.TempFile("pkg/infra/container/imagecontext/base", "dockerfile")
+	case "arm64":
+		tmpFile, errFile = ioutil.TempFile("pkg/infra/container/imagecontext/arm", 	"dockerfile")
+	default:
+		return fmt.Errorf("unsupported architecture: %s", arch)
+	}
+	if errFile != nil {
+		return errFile
+	}
+	defer os.Remove(tmpFile.Name())
+	_, err = tmpFile.Write(newDockerfile)
+	if err != nil {
+		return err 
+	}
+	var cmd *exec.Cmd
+	switch arch {
+	case "amd64":
+		cmd = exec.Command("docker", "build", "-t", imageName, "-f", tmpFile.Name(), "pkg/infra/container/imagecontext/base")
+	case "arm64":
+		cmd = exec.Command("docker", "build", "-t", imageName, "-f", tmpFile.Name(),  "pkg/infra/container/imagecontext/arm")
```

Comment:
`pkg/infra/container/imagecontext/arm` this Dockerfile is long time no updated, so you may first to test whether this can be used as a base-image env for sealer image to run on arm arch. If not you need first modifiy the dockerfile to fix this. In amd arch, you can follow [How to run e2e test locally](https://github.com/sealerio/sealer/blob/main/CONTRIBUTING.md#how-to-run-e2e-test-locally) to run sealer in container locally, it is not supported in arm arch now(this is one reason we need to support run sealer image locally
no matter the arch is what).
So I suggest you to do these:
1. test whether `pkg/infra/container/imagecontext/arm` can be used as a base-image env for sealer image to run on arm arch, you can use https://github.com/sealerio/sealer/blob/dd72089290c8fe31ac0d3670e6704d7f38b457cf/pkg/infra/test/container_test.go#L103
to run some containers as nodes for testing.
2. if not, modifiy the dockerfile to support build a base-image env for sealer image run in arm arch.
3. coding use the arm-base-image for running sealer in container locally in arm arch.
Relevant referable codes are mainly located in /infra,/test directories.

Reply:
Sorry for my late reply, I just recovered from a bad cold.
So my work is to test and modify `pkg/infra/container/imagecontext/arm/Dockerfile` to support run sealer locally in arm arch. Am I understanding you correctly?

Reply:
I'm not quite understanding here. Could you explain that a bit more? Thx.

Reply:
> 1. test whether `pkg/infra/container/imagecontext/arm` can be used as a base-image env for sealer image to run on arm arch, you can use  to run some containers as nodes for testing.
I'm not quite understanding here. Could you explain that a bit more? Thx.

Reply:
> > you can use  [sealer/pkg/infra/test/container_test.go] to run some containers as nodes for testing.
> 
> I'm not quite understanding here. Could you explain that a bit more? Thx.

This code can run some containers as node(use base-image), you can use this code to run the infra and test whether sealer can run on this infra env.

Reply:
> Sorry for my late reply, I just recovered from a bad cold. So my work is to test and modify `pkg/infra/container/imagecontext/arm/Dockerfile` to support run sealer locally in arm arch. Am I understanding you correctly?

Yes~